pkg/master/master.go

```diff
@@ -155,6 +178,52 @@ type completedConfig struct {
 	*Config
 }
 
+func (c *Config) createMasterCountReconciler() EndpointReconciler {
+	// use a default endpoint reconciler if nothing is set
+	endpointClient := coreclient.NewForConfigOrDie(c.GenericConfig.LoopbackClientConfig)
+	return NewMasterCountEndpointReconciler(c.MasterCount, endpointClient)
+}
+
+func (c *Config) createLeaseReconciler() EndpointReconciler {
+	ttl := c.MasterEndpointReconcileTTL
+	config, err := c.StorageFactory.NewConfig(kapi.Resource("apiServerIPInfo"))
+	if err != nil {
+		glog.Fatalf("Error determining service IP ranges: %v", err)
+	}
+	leaseStorage, _, err := storagefactory.Create(*config)
+	if err != nil {
+		glog.Fatalf("Error creating storage factory: %v", err)
+	}
+	endpointConfig, err := c.StorageFactory.NewConfig(kapi.Resource("endpoints"))
+	if err != nil {
+		glog.Fatalf("Error getting storage config: %v", err)
+	}
+	endpointsStorage := endpointsstorage.NewREST(generic.RESTOptions{
+		StorageConfig:           endpointConfig,
+		Decorator:               generic.UndecoratedStorage,
+		DeleteCollectionWorkers: 0,
+		ResourcePrefix:          c.StorageFactory.ResourcePrefix(kapi.Resource("endpoints")),
+	})
+	endpointRegistry := endpoint.NewRegistry(endpointsStorage)
+	masterLeases := election.NewLeases(leaseStorage, "/masterleases/", uint64(ttl))
+	return election.NewLeaseEndpointReconciler(endpointRegistry, masterLeases)
+}
+
+func (c *Config) createEndpointReconciler() EndpointReconciler {
+	glog.Infof("Using reconciler: %v", c.EndpointReconcilerType)
+	switch c.EndpointReconcilerType {
+	case "":
```

Comment:
A option to disable the reconciler, would be useful for people who use self-hosted k8s.

Then they can just create a `kubernetes` service which use selector.

See: https://github.com/kubernetes/community/pull/939#issuecomment-324081192 and https://github.com/kubernetes/community/pull/939#issuecomment-326348679

Reply:
i added a `none` reconciler to not do anything on the reconcile loop.